### PR TITLE
[test] add test case in multi_ail to test peer br counts

### PR DESCRIPTION
**Pull request by zesonzhang**
This new test case ensures under the multi-ail topology, the count of peer BR on AIL (adjcent BR) is less than the count of peer BR seen in the network data.

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_multi_ail.py [hunk 1]
```diff
@@ -124,6 +124,25 @@ def test_multi_backbone_infra(self):
         self.assertTrue(br2.ping(br3_onlink_ula, backbone=True))
         self.assertTrue(br3.ping(br2_onlink_ula, backbone=True))
 
+    def test_adjcent_peer_br_count_less_than_netdata_peer_br_count(self):
+        """This test ensures that the adjcent_peer_br count is less than netdata_peer_br count in Mutil-AIL topology."""
```
Comment (superwhd) [inline tests/scripts/thread-cert/border_router/test_multi_ail.py:128]: > ensures that the adjcent_peer_br count is less than netdata_peer_br count

But there isn't such a check in the test body. Maybe reword this description.

File: tests/scripts/thread-cert/border_router/test_multi_ail.py [hunk 2]
```diff
@@ -124,6 +124,25 @@ def test_multi_backbone_infra(self):
...
+        self.simulator.go(config.BORDER_ROUTER_STARTUP_DELAY)
+
+        self.assertEqual(len(br1.get_br_routers_peer_ip_addresses()), 0)
+        self.assertEqual(len(br1.get_br_peers_rloc16s()), 2)
+
+        self.assertEqual(len(br2.get_br_routers_peer_ip_addresses()), 1)
+        self.assertEqual(len(br2.get_br_peers_rloc16s()), 2)
+
+        self.assertEqual(len(br3.get_br_routers_peer_ip_addresses()), 1)
+        self.assertEqual(len(br3.get_br_peers_rloc16s()), 2)
```
Comment (superwhd) [inline tests/scripts/thread-cert/border_router/test_multi_ail.py:144]: I think it's also worth verifying the output of dbus because that's where the telemetry is collected.

Comment (zesonzhang) [reply tests/scripts/thread-cert/border_router/test_multi_ail.py:144]: Thanks for the suggestions!
There's a plan to test dbus related in ot-br-posix repo, I will focus on the BR count test here.

File: tests/scripts/thread-cert/border_router/test_multi_ail.py (hunk 1 above)
Comment (zesonzhang) [reply tests/scripts/thread-cert/border_router/test_multi_ail.py:128]: Thanks, will update.

File: tests/scripts/thread-cert/border_router/test_multi_ail.py (hunk 2 above)
Comment (superwhd) [reply tests/scripts/thread-cert/border_router/test_multi_ail.py:149]: > There's a plan to test dbus related in ot-br-posix repo.

AFAIK there isn't an easy way to set up multi AIL environment in ot-br-posix. Could you share your idea?

Comment (zesonzhang) [reply tests/scripts/thread-cert/border_router/test_multi_ail.py:149]: initial idea is to use a shell script to create docker network and run docker containers to simulate OTBR nodes
